broker.go

```diff
@@ -118,6 +118,11 @@ func (b *Broker) Open(conf *Config) error {
 		}
 
 		if conf.Net.SASL.Enable {
+			b.connErr = b.sendAndReceiveSASLPlainHandshake()
+			if b.connErr != nil {
+				Logger.Printf("Error while performing SASL handshake %s\n", b.addr)
+				return
```

Comment:
It might be simpler just to call `sendAndReceiveSASLPlainHandshake` from `sendAndReceiveSASLPlainAuth` and re-use the existing error handling for that method?